python/hsfs/core/hudi_engine.py

```diff
@@ -0,0 +1,195 @@
+    HUDI_KEY_GENERATOR_OPT_KEY = "hoodie.datasource.write.keygenerator.class"
+    HUDI_COMPLEX_KEY_GENERATOR_OPT_VAL = "org.apache.hudi.keygen.CustomKeyGenerator"
+    HIVE_PARTITION_EXTRACTOR_CLASS_OPT_KEY = (
+        "hoodie.datasource.hive_sync.partition_extractor_class"
+    )
+    DEFAULT_HIVE_PARTITION_EXTRACTOR_CLASS_OPT_VAL = (
+        "org.apache.hudi.hive.MultiPartKeysValueExtractor"
+    )
+    HUDI_COPY_ON_WRITE = "COPY_ON_WRITE"
+    HUDI_BULK_INSERT = "bulk_insert"
+    HUDI_INSERT = "insert"
+    HUDI_UPSERT = "upsert"
+    HUDI_DELETE = "delete"
+    HUDI_QUERY_TYPE_OPT_KEY = "hoodie.datasource.query.type"
+    HUDI_QUERY_TYPE_SNAPSHOT_OPT_VAL = "snapshot"
+    HUDI_QUERY_TYPE_INCREMENTAL_OPT_VAL = "incremental"
+    HUDI_BEGIN_INSTANTTIME_OPT_KEY = "hoodie.datasource.read.begin.instanttime"
+    HUDI_END_INSTANTTIME_OPT_KEY = "hoodie.datasource.read.end.instanttime"
+    PAYLOAD_CLASS_OPT_KEY = "hoodie.datasource.write.payload.class"
+    PAYLOAD_CLASS_OPT_VAL = "org.apache.hudi.common.model.EmptyHoodieRecordPayload"
+
+    def __init__(
+        self,
+        feature_store_id,
+        feature_store_name,
+        feature_group,
+        spark_context,
+        spark_session,
+    ):
+        self._feature_group = feature_group
+        self._spark_context = spark_context
+        self._spark_session = spark_session
+        self._feature_store_id = feature_store_id
+        self._feature_store_name = feature_store_name
+        self._base_path = self._feature_group.location
+        self._table_name = feature_group.name + "_" + str(feature_group.version)
+
+        # TODO (Davit): Need to fix ComplexKeyGenerator bug
+        self._primary_key = feature_group.primary_key[0]
+        self._partition_key = feature_group.partition_key[0]
+        self._precombine_key = feature_group.partition_key[0]
+
+        self._feature_group_api = feature_group_api.FeatureGroupApi(feature_store_id)
+        self._storage_connector_api = storage_connector_api.StorageConnectorApi(
+            self._feature_store_id
+        )
+        self._connstr = self._storage_connector_api.get(
+            self._feature_store_name, storage_connector.StorageConnector.JDBC,
+        ).connection_string
+
+    def save_hudi_fg(self, dataset, save_mode, operation, write_options):
+        fg_commit = self._write_hudi_dataset(
+            dataset, save_mode, operation, write_options
+        )
+
+        return self._feature_group_api.commit(self._feature_group, fg_commit)
+
+    def register_temporary_table(self, alias, start_timestamp, end_timestamp):
+        self._spark_session.conf.set("spark.sql.hive.convertMetastoreParquet", "false")
+
+        hudi_options = self._setup_hudi_read_opts(start_timestamp, end_timestamp)
+        self._spark_session.read.format(self.HUDI_SPARK_FORMAT).options(
+            **hudi_options
+        ).load(self._base_path).registerTempTable(alias)
```

Comment:
`registerTempTable` is deprecated, you should use `createOrReplaceTempView` instead. Which also takes care of overwriting the alias in case you run multiple queries in the same Spark session.

Reply:
done